### PR TITLE
fix(static-analysis): unpausing and refractoring semgrep checks

### PR DESCRIPTION
**Pull request by pkanoongo**
**Description**

The PR restructures the repo to move the semgrep workflow action from '/static-analysis/semgrep' to '/static-analysis'. All the references to the semgrep are updated to use the new path. Also, unpausing the semgrep checks.

**Fixes**

- fix(static-analysis):restructuring the repos.
- fix(static-analysis): test unpausing semgrep checks

### CODE REVIEW

**Review by prathamesh16c — CHANGES_REQUESTED** [review]
can you update the path on the test file too

**Comment by pkanoongo** [comment]
Update the path on test. Both checks have passed. Thanks!!

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -21,6 +21,6 @@ jobs:
       - uses: open-turo/actions-gha/test@v2
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: ./static-analysis/semgrep
+      - uses: ./static-analysis
```
Comment (pkanoongo) [inline .github/workflows/ci.yaml:24]: Updated the path here.

**Review by pkanoongo — COMMENTED** [review]
Fixed the path.

**Review by prathamesh16c — CHANGES_REQUESTED** [review]
@pkanoongo please can you create a retroactive breaking change doc in this repo and add instructions for the consuming repos on using the refactored path of this action.

File: static-analysis/action.yaml
[the remark this answers is not in the document]
Comment (pkanoongo) [reply static-analysis/action.yaml:1]: I think option2 is better. We will add instructions in our breaking change doc.

**Comment by pkanoongo** [comment]
Cleanup:
This is no longer required. Closing the PR.

**Pull request closed without merging**